cmd/bucket-handlers-listobjects.go

```diff
@@ -52,6 +52,41 @@ func validateListObjectsArgs(prefix, marker, delimiter string, maxKeys int) APIE
 	return ErrNone
 }
 
+func checkRequestAuthType(w http.ResponseWriter, r *http.Request, bucket, policyAction, region string) APIErrorCode {
+	reqAuthType := getRequestAuthType(r)
+
+	switch reqAuthType {
+	case authTypePresignedV2, authTypeSignedV2:
+		// Signature V2 validation.
+		s3Error := isReqAuthenticatedV2(r)
+		if s3Error != ErrNone {
+			errorIf(errSignatureMismatch, dumpRequest(r))
+			writeErrorResponse(w, r, s3Error, r.URL.Path)
+		}
+		return s3Error
+	case authTypeSigned, authTypePresigned:
+		s3Error := isReqAuthenticated(r, region)
+		if s3Error != ErrNone {
+			errorIf(errSignatureMismatch, dumpRequest(r))
+			writeErrorResponse(w, r, s3Error, r.URL.Path)
+		}
+		return s3Error
+	}
+
+	if reqAuthType == authTypeAnonymous && policyAction != "" {
+		// http://docs.aws.amazon.com/AmazonS3/latest/dev/using-with-s3-actions.html
+		s3Error := enforceBucketPolicy(bucket, policyAction, r.URL)
+		if s3Error != ErrNone {
+			writeErrorResponse(w, r, s3Error, r.URL.Path)
+		}
+		return s3Error
+	}
+
+	// By default return ErrAccessDenied
+	writeErrorResponse(w, r, ErrAccessDenied, r.URL.Path)
+	return ErrAccessDenied
```

Comment:
Yes. The function returning boolean is good enough. As I am still reading other handlers, I made it as interim. I will fix it in upcoming update of this PR.